### PR TITLE
1704: Update in registration of custom resource providers

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/opensrp/fhircore/issues/1704

- [x] Update the way of registration of custom resource providers
- [x] Bump up versions of the custom extensions module

### CODE REVIEW

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 1]
```diff
@@ -448,5 +462,45 @@ public static IServerConformanceProvider<?> calculateConformanceProvider(IFhirSy
 			throw new IllegalStateException();
 		}
 	}
+
+	private void registerLocationHierarchyTypes(RestfulServer restfulServer, DaoRegistry daoRegistry) {
```
Comment (dubdabasoduba) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java:466]: Add the Hardcoded strings to Constants or use the existing Constants.

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java [hunk 2]
```diff
@@ -448,5 +462,45 @@ public static IServerConformanceProvider<?> calculateConformanceProvider(IFhirSy
...
+		restfulServer.getFhirContext().registerCustomType(ChildTreeNode.class);
+	}
+
+	private void registerPracitionerDetailsTypes(RestfulServer restfulServer, DaoRegistry daoRegistry) {
+		IFhirResourceDao<Practitioner> practitionerIFhirResourceDao = daoRegistry.getResourceDao("Practitioner");
+		IFhirResourceDao<PractitionerRole> practitionerRoleIFhirResourceDao = daoRegistry.getResourceDao("PractitionerRole");
+		IFhirResourceDao<CareTeam> careTeamIFhirResourceDao = daoRegistry.getResourceDao("CareTeam");
+		IFhirResourceDao<OrganizationAffiliation> organizationAffiliationIFhirResourceDao = daoRegistry.getResourceDao("OrganizationAffiliation");
+		IFhirResourceDao<Organization> organizationIFhirResourceDao = daoRegistry.getResourceDao("Organization");
+		IFhirResourceDao<Location> locationIFhirResourceDao = daoRegistry.getResourceDao("Location");
```
Comment (dubdabasoduba) [inline src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java:487]: Add the Hardcoded strings to Constants or use the existing Constants.

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java:467]: Done

File: src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/ca/uhn/fhir/jpa/starter/common/StarterJpaConfig.java]: Done

**Review by dubdabasoduba — APPROVED** [review]

**Review by dubdabasoduba — APPROVED** [review]

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**